vault/diagnose/helpers_test.go

```diff
@@ -63,27 +85,51 @@ func TestDiagnoseOtelResults(t *testing.T) {
 const coffeeLeft = 3
 
 func makeCoffee(ctx context.Context) error {
+
 	if coffeeLeft < 5 {
 		Warn(ctx, "coffee getting low")
 		Advise(ctx, getMoreCoffee)
 	}
 
-	err := Test(ctx, "warm-milk", warmMilk)
-	if err != nil {
-		return err
-	}
+	// To mimic listener TLS checks, we'll see if we can nest a Test and add errors in the function
+	Test(ctx, "prepare-kitchen", func(ctx context.Context) error {
+		return Test(ctx, "build-microwave", func(ctx context.Context) error {
+			buildMicrowave(ctx)
+			return nil
+		})
+	})
 
-	err = brewCoffee(ctx)
-	if err != nil {
-		return err
-	}
+	Test(ctx, "warm-milk", func(ctx context.Context) error {
+		return warmMilk(ctx)
+	})
+
+	brewCoffee(ctx)
 
 	SpotCheck(ctx, "pick-scone", pickScone)
 
 	Test(ctx, "dispose-grounds", Skippable("dispose-grounds", disposeGrounds))
 	return nil
 }
 
+// buildMicrowave will throw an error in the function itself to fail the span,
+// but will return nil so the caller test doesn't necessarily throw an error.
+// The intended behavior is that the superspan will detect the failed subspan
+// and fail regardless.This happens when Fail is used to fail the span, but not
+// when Error is used. See the comment in the function itself.
+func buildMicrowave(ctx context.Context) error {
+	ctx, span := StartSpan(ctx, "buy-parts")
+
+	Fail(ctx, "no stores sell microwave parts, please buy a microwave instead.")
+
+	// The error line here does not actually yield an error in the output.
+	// TODO: Debug this. In the meantime, always use Fail over Error.
```

Comment:
Does this TODO need to be addressed?

Reply:
It's not urgent since we can and are just avoiding the use of `Error` for now, but we'd need to fix it eventually. I could put a `deprecated` warning above the Error method, but I figure we'll be fixing it post code-freeze, so it's fine.